### PR TITLE
update simulation when loading from web

### PR DESCRIPTION
**Pull request by tylerflex**
This is just a first stab at some updater improvements.  I figured it would be best to get this done before I lose internet in an hour.

Feel free to test, merge, or just wait.  I've tested upgrading from 1.3 -> 1.4 and also loading 1.4 -> 1.4.

### CODE REVIEW

**Comment by momchil-flex** [comment]
Sorry, I know I was talking about `web.load` yesterday, but I think the conversion should actually be already incorporated in the `SimulationData.from_file` level. And honestly I don't see why not even incorporate it by overwriting `Simulation.parse_obj` and `Simulation.from_file`...?

**Comment by tylerflex** [comment]
Yea I think that makes more sense.  I'll make changes today

File: tidy3d/components/simulation.py [hunk 1]
```diff
@@ -1742,3 +1743,30 @@ def make_eps_data(coords: Coords):
...
+            An instance of the component class calling `load`.
+
+        Example
+        -------
+        >>> simulation = Simulation.from_file(fname='folder/sim.json')
+        """
+        try:
+            updater = Updater.from_file(fname)
+            sim_dict = updater.update_to_current()
+            return Simulation.parse_obj(sim_dict, **parse_kwargs)
```
Comment (momchil-flex) [inline tidy3d/components/simulation.py:1770]: `log.warning("Could not update simulation to latest version! Trying to load as is.")` or something?

**Comment by momchil-flex** [comment]
Can we replace these lines with a direct "from_file"? 

https://github.com/flexcompute/tidy3d/blob/a01a76dc43187b30660ad50a6a0b5acdb432341e/tests/test_IO.py#L174-L180

**Comment by tylerflex** [comment]
fixed both comments.  Also, see my question about the version number item in the issue: #399

**Review by momchil-flex — APPROVED** [review]

**Pull request merged**